packages/bee-q/src/components/notification/bq-notification.tsx

```diff
@@ -0,0 +1,188 @@
+ */
+@Component({
+  tag: 'bq-notification',
+  styleUrl: './scss/bq-notification.scss',
+  shadow: true,
+})
+export class BqNotification {
+  // Own Properties
+  // ====================
+
+  // Reference to host HTML element
+  // ===================================
+
+  @Element() el!: HTMLBqNotificationElement;
+
+  // State() variables
+  // Inlined decorator, alphabetical order
+  // =======================================
+
+  @State() private isHidden = true;
+
+  // Public Property API
+  // ========================
+
+  /** Type of Notification */
+  @Prop({ reflect: true }) type: TNotificationType = 'default';
+
+  /** Set property if you want Notification icon to be shown. */
+  @Prop({ reflect: true }) showIcon: boolean;
+
+  /** Set the subject color if you don't want to be black. Subject color will also apply to Icon color if there is one. */
+  @Prop({ reflect: true }) subjectColor: string;
+
+  /** Set property to false if you want to hide Close icon */
+  @Prop({ reflect: true }) showClose: boolean;
+
+  // Prop lifecycle events
+  // =======================
+  @Watch('type')
+  checkPropValues() {
+    validatePropValue(NOTIFICATION_TYPES, 'default', this.el, 'type');
+  }
+
+  // Events section
+  // Requires JSDocs for public API documentation
+  // ==============================================
+
+  // Component lifecycle events
+  // Ordered by their natural call order
+  // =====================================
+
+  componentWillLoad() {
+    this.checkPropValues();
+  }
+
+  // Listeners
+  // ==============
+
+  // Public methods API
+  // These methods are exposed on the host element.
+  // Always use two lines.
+  // Public Methods must be async.
+  // Requires JSDocs for public API documentation.
+  // ===============================================
+
+  /**
+   * Trigger function when you want to close Notification
+   */
+  @Method()
+  async hide() {
+    this.hideNotification();
+  }
+
+  /**
+   * Trigger function when you want to show Notification
+   */
+  @Method()
+  async show() {
+    this.showNotification();
+  }
+
+  // Local methods
+  // Internal business logic.
+  // These methods cannot be called from the host element.
+  // =======================================================
+
+  private getNotificationIconName = () => {
+    if (!this.showIcon) return '';
+
+    const type = this.type;
+    let icon = 'info';
+
+    if (type === 'success') {
+      icon = 'check-circle';
+    }
+
+    if (type === 'error') {
+      icon = 'x-circle';
+    }
+
+    return icon;
+  };
+
+  private getNotificationColor = () => {
+    const type = this.type;
+    const colors = {
+      default: 'ui--inverse-active',
+      success: 'ui--success',
+      info: 'ui--brand',
+      warning: 'ui--warning',
+      error: 'ui--danger',
+    };
+
+    return isDefined(this.subjectColor) ? this.subjectColor : colors[type];
+  };
```

Comment:
We can leverage on CSS to set the icon color based on the notification type:

If we can have something like:

```jsx
{/* ICON */}
{!this.hideIcon && (
  <div
    class={{
      'notification--icon mr-2 inline-block text-left align-top': true,
      [`color-${this.type}`]: true, // The icon color will be based on the type (info, success, warning, error)
    }}
  >
    {!this.hasCustomIcon ? <bq-icon name={this.iconName} /> : <slot name="icon" />}
  </div>
)}
```
then we can do the following with SCSS
```scss
/**
 * Set the nocitication icon color based on the type value selected
 */
.notification--icon.color {
  &-default {
    @apply text-text-primary;
  }

  &-info {
    @apply text-text-brand;
  }

  &-error {
    @apply text-text-danger;
  }

  &-success {
    @apply text-text-success;
  }

  &-warning {
    @apply text-text-warning;
  }
}
```